libgeopmd/src/RolloverGenerator.cpp

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2015 - 2025 Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <string>
+#include <cmath>
+#include "RolloverGenerator.hpp"
+#include "geopm/Exception.hpp"
+
+namespace geopm
+{
+    RolloverGenerator::RolloverGenerator(int bit_width)
+        : m_last_value(0.0)
+        , m_rollover_factor(pow(2, bit_width))
```

Comment:
Why set ``m_rollover_factor`` in the initializer list, and also provide the ``set_factor()`` method?  Why both?